### PR TITLE
Unify GlobalTrajectoryBuilder across 2D and 3D

### PR DESCRIPTION
**Pull request by cschuet**
This PR unifies GlobalTrajectoryBuilder across 2D and 3D by templating
it with the LocalTrajectoryBuilder and the SparsePoseGraph. It thus
includes small changes to interfaces in LocalTrajectoryBuilder and
SparsePoseGraph.

### CODE REVIEW

File: cartographer/mapping_3d/local_trajectory_builder.cc [hunk 1]
```diff
@@ -58,9 +58,7 @@ void LocalTrajectoryBuilder::AddImuData(const sensor::ImuData& imu_data) {
 }
 
 std::unique_ptr<LocalTrajectoryBuilder::InsertionResult>
-LocalTrajectoryBuilder::AddRangefinderData(const common::Time time,
-                                           const Eigen::Vector3f& origin,
-                                           const sensor::PointCloud& ranges) {
+LocalTrajectoryBuilder::AddRangeData(const common::Time time,                                           const sensor::RangeData& range_data) {
```
Comment (wohe) [inline cartographer/mapping_3d/local_trajectory_builder.cc]: Run `clang-format`?

File: cartographer/mapping/global_trajectory_builder.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+namespace mapping {
+
+template <typename LocalTrajectoryBuilder,
+          typename LocalTrajectoryBuilderOptions, typename SparsePoseGraph>
+class GlobalTrajectoryBuilder
+    : public mapping::GlobalTrajectoryBuilderInterface {
+ public:
+  GlobalTrajectoryBuilder(const LocalTrajectoryBuilderOptions& options,
+                          const int trajectory_id,
+                          SparsePoseGraph* sparse_pose_graph)
```
Comment (wohe) [inline cartographer/mapping/global_trajectory_builder.h]: `SparsePoseGraph* const`?

File: cartographer/mapping/global_trajectory_builder.h [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  ~GlobalTrajectoryBuilder() override {}
+
+  GlobalTrajectoryBuilder(const GlobalTrajectoryBuilder&) = delete;
+  GlobalTrajectoryBuilder& operator=(const GlobalTrajectoryBuilder&) = delete;
+
+  const mapping::PoseEstimate& pose_estimate() const override {
+    return local_trajectory_builder_.pose_estimate();
+  }
+
+  void AddRangefinderData(common::Time time, const Eigen::Vector3f& origin,
```
Comment (wohe) [inline cartographer/mapping/global_trajectory_builder.h]: `const common::Time`?

File: cartographer/mapping_3d/local_trajectory_builder.cc (hunk 1 above)
Comment (cschuet) [inline cartographer/mapping_3d/local_trajectory_builder.cc]: done

File: cartographer/mapping/global_trajectory_builder.h (hunk 2 above)
Comment (cschuet) [inline cartographer/mapping/global_trajectory_builder.h]: done

**Pull request merged**